src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c

```diff
@@ -21,47 +48,601 @@ const ActorInit Bg_Po_Event_InitVars = {
+        if (Flags_GetSwitch(globalCtx, thisx->params)) {
+            Actor_Kill(&this->dyna.actor);
+        } else {
+            BgPoEvent_InitPaintings(this, globalCtx);
+        }
+    } else {
+        DynaPolyInfo_SetActorMove(&this->dyna, 0);
+        if (Flags_GetSwitch(globalCtx, thisx->params)) {
+            Actor_Kill(&this->dyna.actor);
+        } else {
+            BgPoEvent_InitBlocks(this, globalCtx);
+        }
+    }
+}
+
+void BgPoEvent_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgPoEvent* this = THIS;
+
+    if (this->actorType >= 2) {
+        Collider_DestroyTris(globalCtx, &this->collider1);
+    } else {
+        DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+        if ((this->actorType == 1) && (gSaveContext.timer1Value > 0)) {
+            gSaveContext.timer1State = 0xA;
+        }
+    }
+}
+
+void BgPoEvent_BlockWait(BgPoEvent* this, GlobalContext* globalCtx) {
+
+    this->dyna.actor.posRot.pos.y = 833.0f;
+    if (puzzleState == 0x3F) {
+        if (this->actorType == 1) {
+            func_800800F8(globalCtx, 0xC4E, 0x41, NULL, 0);
+        }
+        this->timer = 45;
+        this->actionFunc = BgPoEvent_BlockShake;
+    } else if (this->dyna.actor.xzDistFromLink > 50.0f) {
+        if (this->actorType != 1) {
+            puzzleState |= (1 << this->actorIndex);
+        } else {
+            puzzleState |= 0x10;
+        }
+    } else if (this->actorType != 1) {
+        puzzleState &= ~(1 << this->actorIndex);
+    } else {
+        puzzleState &= ~0x10;
+    }
+}
+
+void BgPoEvent_BlockShake(BgPoEvent* this, GlobalContext* globalCtx) {
+
+    DECR(this->timer);
+    if (this->timer < 15) {
+        this->dyna.actor.posRot.pos.x = this->dyna.actor.initPosRot.pos.x + 2.0f * ((this->timer % 3) - 1);
+        if (!(this->timer % 4)) {
+            Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_BLOCK_SHAKE);
+        }
+    }
+    if (this->timer == 0) {
+        this->dyna.actor.posRot.pos.x = this->dyna.actor.initPosRot.pos.x;
+        puzzleState = 0;
+        this->timer = 60;
+        this->actionFunc = BgPoEvent_BlockFall;
+    }
+}
+
+void BgPoEvent_BlockCheck(BgPoEvent* this) {
+    s32 temp_f10;
+    s32 temp_f6;
+    s32 phi_v1;
+    s32 phi_a1;
+    s32 phi_t0;
+    s32 phi_a3;
+
+    if ((this->actorIndex == 3) || (this->actorIndex == 1)) {
+        temp_f6 = this->dyna.actor.posRot.pos.z;
+        temp_f10 = this->dyna.actor.child->posRot.pos.z;
+        if (this->actorIndex == 3) {
+            phi_v1 = temp_f6;
+            phi_a1 = temp_f10;
+            phi_a3 = this->dyna.actor.posRot.pos.x;
+            phi_t0 = this->dyna.actor.child->posRot.pos.x;
+        } else { // this->actorIndex == 1
+            phi_v1 = temp_f6;
+            phi_a1 = temp_f10;
+            phi_a3 = this->dyna.actor.child->posRot.pos.x;
+            phi_t0 = this->dyna.actor.posRot.pos.x;
+        }
```

Comment:
this is kinda weird, can you set  `phi_v1` and `phi_a1` outside of the sub-if/else,
or alternatively get rid of temp_f6 and temp_f10?